machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,184 @@
+
+	// systemType is the System type used to host the instance.
+	// systemType determines the number of cores and memory that is available.
+	// Few of the supported SystemTypes are s922,e880,e980.
+	// e880 systemType available only in Dallas Datacenters.
+	// e980 systemType available in Datacenters except Dallas and Washington.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The current default is s922 which is generally available.
+	// + This is not an enum because we expect other values to be added later which should be supported implicitly.
+	// +optional
+	SystemType string `json:"systemType,omitempty"`
+
+	// processorType is the VM instance processor type.
+	// It must be set to one of the following values: Dedicated, Capped or Shared.
+	// Dedicated: resources are allocated for a specific client, The hypervisor makes a 1:1 binding of a partition’s processor to a physical processor core.
+	// Shared: Shared among other clients.
+	// Capped: Shared, but resources do not expand beyond those that are requested, the amount of CPU time is Capped to the value specified for the entitlement.
+	// if the processorType is selected as Dedicated, then processors value cannot be fractional.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The current default is Shared.
+	// +kubebuilder:validation:Enum:="Dedicated";"Shared";"Capped"
+	// +optional
+	ProcessorType string `json:"processorType,omitempty"`
+
+	// processors is the number of virtual processors in a virtual machine.
+	// when the processorType is selected as Dedicated the processors value cannot be fractional.
+	// maximum value for the Processors depends on the selected SystemType.
+	// when SystemType is set to e880 or e980 maximum Processors value is 143.
+	// when SystemType is set to s922 maximum Processors value is 15.
+	// minimum value for Processors depends on the selected ProcessorType.
+	// when ProcessorType is set as Shared or Capped, The minimum processors is 0.5.
+	// when ProcessorType is set as Dedicated, The minimum processors is 1.
+	// When omitted, this means that the user has no opinion and the platform is left to choose a
+	// reasonable default. The default is set based on the selected ProcessorType.
+	// when ProcessorType selected as Dedicated, the default is set to 1.
+	// when ProcessorType selected as Shared or Capped, the default is set to 0.5.
+	// +optional
+	Processors intstr.IntOrString `json:"processors,omitempty"`
+
+	// memoryGiB is the size of a virtual machine's memory, in GiB.
+	// maximum value for the MemoryGiB depends on the selected SystemType.
+	// when SystemType is set to e880 maximum MemoryGiB value is 6950.4604 GiB.
+	// when SystemType is set to e980 maximum MemoryGiB value is 14255.755 GiB.
+	// when SystemType is set to s922 maximum MemoryGiB value is 14847.144 GiB.
```

Comment:
Where did these values come from? When I read the specifications it said s922 had a maximum of 4TiB of memory or 4096 GiB? Likewise e980 said 64TiB or 65536 GiB. e880 is marketed as 32TiB or 32786 GiB.

Reply:
I took that data from here: https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-pricing-virtual-server, Also I need to update the value for s922 to 877.306 GiB.

Reply:
I believe the values in their documentation are actually already in GiB, in which case we should just copy those values across. That said, I'm not sure whether we want to go with these maximums or the specified maximums for the IBM server types? Have you seen their spec sheets before? 

Eg [from here](https://www.ibm.com/common/ssi/ShowDoc.wss?docURL=/common/ssi/rep_sm/1/897/ENUS9009-_h01/index.html&request_locale=en)

> The Power S922 server supports a maximum of 32 DDR4 DIMM slots. Memory features supported are 8 GB, 16 GB, 32 GB, 64 GB, and 128 GB and run at different speeds of 2133, 2400, and 2666 Mbps, offering a maximum system memory of 4096 GB.

Note these also mean GiB but, sigh, they haven't used the correct term

Reply:
oh yeah, I got confused with the units, Here is the summary of values from both the sites

SystemType,  FromPricingDoc ,    FromServerSpec
e880 ,            7,463 GB ,               16 TB  
e980 ,            15,307 GB,              
s922,              942 GB ,                 4TB

Reply:
Have you tried to create, for example, a s922 with over 942 GB? Does it work? Is there anyone from IBM who can recommend which of these we should stick to?

Reading through the pricing doc it infers these maximums are inferred, so I think maybe we go with the pricing maximums